### PR TITLE
Remove more sdist files

### PR DESCRIPTION
**Pull request by ajfriend**
I don't think there's any need for the following files to be included in the Python SDist:

```
src/h3lib/scripts
src/h3lib/examples
src/h3lib/src/apps
```

@isaacbrodsky, could you confirm that's true?

This will further reduce the SDist size (beyond the huge gains of #227), and may also decrease the build time (?).

However the build still fails, but I'm hoping that some small changes to the cmake files can avoid the need for these files?

### CODE REVIEW

**Comment by ajfriend** [comment]
I tried setting

```
set(BUILD_FUZZERS OFF CACHE BOOL "" FORCE)
set(ENABLE_LIBFUZZER OFF CACHE BOOL "" FORCE)
set(ENABLE_COVERAGE OFF CACHE BOOL "" FORCE)
```

but that didn't seem to help.

**Comment by kylebarron** [comment]
I think when I was stripping sdist size in https://github.com/uber/h3-py/pull/227, it didn't build out of the box if I removed those files. I figured the further gains were marginal so I didn't look more closely at how to make the build work after removing those.

**Comment by isaacbrodsky** [comment]
Was the error the same when BUILD_FUZZERS was OFF? Looking at CMakeLists.txt my suspicion is that the targets for executables are disabled, but other references to the source files remain, for example for autoformatting.

**Comment by ajfriend** [comment]
> Was the error the same when BUILD_FUZZERS was OFF? Looking at CMakeLists.txt my suspicion is that the targets for executables are disabled, but other references to the source files remain, for example for autoformatting.

Getting different error output when I set BUILD_FUZZERS to OFF in the diff above.

**Comment by ajfriend** [comment]
OK, cool. Let's just turn off autoformatting. I don't think there's a need for that at this stage. How would we configure that?

**Comment by ajfriend** [comment]
> OK, cool. Let's just turn off autoformatting. I don't think there's a need for that at this stage. How would we configure that?

nvm. Just discovered the `cmake -LA` command. :)

**Comment by isaacbrodsky** [comment]
There's a reference at https://h3geo.org/docs/next/core-library/compilation-options

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,18 +7,22 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
...
+turn_off(BUILD_BENCHMARKS)
+turn_off(BUILD_FILTERS)
+turn_off(BUILD_FUZZERS)
+turn_off(BUILD_GENERATORS)
+turn_off(BUILD_TESTING)
+turn_off(ENABLE_COVERAGE)
+turn_off(ENABLE_DOCS)
+turn_off(ENABLE_FORMAT)
+turn_off(ENABLE_LIBFUZZER)
+turn_off(ENABLE_LINTING)
```
Comment (kylebarron) [inline CMakeLists.txt:25]: Just curious, can you compare compilation times before and after this change? Does this make it a lot faster?

**Review by kylebarron — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,11 +7,22 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
...
-set(BUILD_FILTERS OFF CACHE BOOL "" FORCE)
-set(BUILD_GENERATORS OFF CACHE BOOL "" FORCE)
-set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
-set(ENABLE_DOCS OFF CACHE BOOL "" FORCE)
+# See all options with `cmake -LA` in an `h3/build` directory,
+# or at https://h3geo.org/docs/next/core-library/compilation-options/
+macro(turn_off option_name)
+    set(${option_name} OFF CACHE BOOL "" FORCE)
+endmacro()
+turn_off(BUILD_ALLOC_TESTS)
```
Comment (isaacbrodsky) [inline CMakeLists.txt:15]: BUILD_ALLOC_TESTS should not have any effect when BUILD_TESTING is off. Harmless to disable it though.

**Review by isaacbrodsky — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (ajfriend) [reply CMakeLists.txt:15]: I had a suspicion that that was true, but yeah, figured it couldn't hurt to just disable everything to make sure.

File: CMakeLists.txt (hunk 1 above)
Comment (ajfriend) [reply CMakeLists.txt]: I spot checked some other compilation times. There's *maybe* a gain. I'll try some runs here for a clearer comparison.

Comment (ajfriend) [reply CMakeLists.txt:25]: Seems to be faster, based on the last two runs I looked at, but hard to tell how much is random variation.

**Old config**
Screen Shot 2022-06-06 at 3 51 11 PM

**New config**
Screen Shot 2022-06-06 at 3 51 40 PM

**Pull request merged**